shared/src/lib.rs

```diff
@@ -53,6 +53,18 @@ impl FromStr for H160Wrapper {
     }
 }
 
+/// Wraps H256 with FromStr and Deserialize that can handle a `0x` prefix.
+#[derive(Debug, Deserialize)]
+#[serde(transparent)]
+pub struct H256Wrapper(#[serde(with = "h256_hexadecimal")] pub H256);
+impl FromStr for H256Wrapper {
```

Comment:
Is this type also really necessary? From what I understand, there already exists `impl FromStr for H256` and it does exactly this.

Reply:
Could very well be... I will see if I can remove it.